rust_src/src/xdisp.rs

```diff
@@ -1,13 +1,30 @@
 //! Display generation from window structure and buffer text.
 
-// Remove the feature check when this file gains functions
-// that are not guarded by it.
-#[cfg(feature = "glyph-debug")]
 use remacs_macros::lisp_fn;
 
+use crate::lisp::LispObject;
+use crate::remacs_sys::{bset_update_mode_line, Qnil};
+use crate::threads::ThreadState;
 #[cfg(feature = "glyph-debug")]
 use crate::{interactive::InteractiveNumericPrefix, remacs_sys::trace_redisplay_p};
 
+/// Mark the current buffer for redisplay.
+/// This function may be passed to `add-variable-watcher'.
+#[lisp_fn]
+pub fn set_buffer_redisplay(
+    _symbol: LispObject,
+    _newval: LispObject,
+    _op: LispObject,
+    _where: LispObject,
+) -> LispObject {
+    let mut current_buffer = ThreadState::current_buffer_unchecked();
+    unsafe {
+        bset_update_mode_line(current_buffer.as_mut());
+    }
+    current_buffer.set_prevent_redisplay_optimizations_p(true);
+    Qnil
```

Comment:
You can drop the Qnil and let the function return nothing.

Reply:
Cheers. I have made this change.